Revise response content

mattdavis0351 opened the pull request:
This pull request addresses content feedback from conversations with @brianamarie.  

@hectorsector there are minor changes to the `config.yml` file in this pull request.  They may affect the error handling you are doing.

Can we work together on making sure we can incorporate both changes without a ton of rework?

mattdavis0351 committed:
add labels to before step

mattdavis0351 committed:
move prerequisites from issue to course details

mattdavis0351 committed:
move preparing repo from issue to activity

mattdavis0351 committed:
remove oddly placed explaination

mattdavis0351 committed:
add workflow naming disclaimer

mattdavis0351 committed:
use commit message to populate pull request

mattdavis0351 committed:
change which pull request title is listened for

mattdavis0351 committed:
add sample pull request body

mattdavis0351 committed:
remove unmaintainable image

mattdavis0351 committed:
refactor explain-workflow

mattdavis0351 committed:
add link for more information

mattdavis0351 committed:
add link to actions tab

mattdavis0351 committed:
add question answer

mattdavis0351 committed:
clarify which line to change

mattdavis0351 committed:
update issue name for pull request responses

mattdavis0351 committed:
update issue name

mattdavis0351 committed:
replace initial wokflow name with new pr name

mattdavis0351 committed:
make dev-env more concise

mattdavis0351 committed:
rename first project issue

mattdavis0351 committed:
add step for explain-action

mattdavis0351 committed:
add explain-actions

mattdavis0351 committed:
update explain-metadata

mattdavis0351 committed:
make create-metadata more concise

mattdavis0351 committed:
update explain-runs

mattdavis0351 committed:
update 06_main-js.md

mattdavis0351 committed:
update 07_hello-result.md

mattdavis0351 committed:
update 07_add-input-params.md

mattdavis0351 committed:
update input-params files

mattdavis0351 committed:
add youtbe link and emoji word

mattdavis0351 committed:
09_new-action.md

mattdavis0351 committed:
update 10_action-two.md

mattdavis0351 committed:
update 09_jokeing-around.md

mattdavis0351 committed:
update 10_create-metadata.md

mattdavis0351 committed:
update 10_create-js-files

mattdavis0351 committed:
update 12_trigger-workflow.md

mattdavis0351 committed:
update 13_action-three-workflow.md

mattdavis0351 committed:
update 14_action-three.md

mattdavis0351 committed:
update 14_create-metadata.md

mattdavis0351 committed:
update runs: parameter

mattdavis0351 committed:
update 14_create-js-files.md

mattdavis0351 committed:
update 14_js-files-activity

mattdavis0351 committed:
update 15_add-a-label

brianamarie commented on course-details.md:
```diff
@@ -3,4 +3,14 @@
 Over the duration of this course, approximately 1 hour, you will learn the skills needed to begin using and customizing GitHub Actions in your unique workflow scenarios.
 
 **prerequisites**
```
```suggestion
**Prerequisites**
```

brianamarie committed:
Correct button name for first workflow creation

brianamarie commented on responses/01_welcome-activity.md:
```diff
@@ -1,10 +1,35 @@
+   - Change the name of the file from `main.yml` to `my-workflow.yml`.
+1. Commit the workflow to a new branch named `add-initial-workflow`.
+1. Create a pull request titled **Create my-workflow.yml**.
+1. Supply the pull request body content and click `Commit new file`.
```
I edited this and now I realize I did the wrong thing 🙈 I'll fix it! I'm sorry!

mattdavis0351 replied:
I forgive you!

brianamarie committed:
revert last commit

brianamarie commented on responses/01_welcome-activity.md:
```diff
@@ -1,10 +1,35 @@
+
+</details>
+
+I'll respond in the new pull request when I detect it has been created.
```
@mattdavis0351 I think it would be a good idea to cover your bases and add a response to this issue pointing to the new PR in addition to the new comments in the newly opened PR.

mattdavis0351 replied:
Will add in iteration 2 for this.  Since this doesn't break anything and is more of a creature comfort thing I think we can wait to add that until we staff ship this and get more feedback.

Sound fair?

brianamarie replied:
👍 Doesn't matter to me when it gets added, as long as it does. My personal preference as a course author is to add this stuff when it comes up so that I don't forget it later, but that's just my preference. As long as there's a way of making sure this feedback doesn't get lost before the final ship, it's good with me.

mattdavis0351 replied:
I also don't want to make the `config.yml` edits harder on @hectorsector.  I think we can address this piece in his PR when he does the error handling.

brianamarie committed:
standardize punctuation in ordered list

brianamarie reviewed (commented):
Hi @mattdavis0351! I started going through this, and found just a few tiny things so far. 🎉 But, I did get stuck after the first step. I'm guessing that one of the posted responses should give me instructions for my next step, but I only saw the more informational comments: https://github.com/brianamarie/writing-javascript-actions/pull/2

I'm wondering if the event on line 69 didn't get triggered, or if it did, and the gate failed, so  02_workflow-results.md didn't post?

https://github.com/githubtraining/write-javascript-actions/blob/matt-revisions/config.yml#L69-L90

mattdavis0351 committed:
add debuggin step to troubleshoot

mattdavis0351 committed:
Merge branch 'matt-revisions' of https://github.com/githubtraining/write-javascript-actions into matt-revisions

add debugging for troubleshooting

mattdavis0351 committed:
Update course-details.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 commented:
@brianamarie the first step getting stuck is not unique to this course.  @hectorsector and @JasonEtco have explained that it has something to with the draft not syncing as we expect.  It is happening in other courses as well.

mattdavis0351 committed:
rearrange steps for timing puposes

mattdavis0351 committed:
Update 10_action-two.md

mattdavis0351 committed:
Update 14_action-three.md

mattdavis0351 committed:
Update 14_create-metadata.md

mattdavis0351 commented:
@brianamarie @hectorsector I was able to get through all of the content on a test run.  Made a few more minor edits along the way. 

The current concern is how different this is from the error handling fixes taking place here: #6 

This is the content that needs to make it into the course, I made a few changes to this config which is where the bigger concern is.

Let's figure out how to get these changes in without massive conflicts!

hectorsector commented:
I suggest merging whichever PR is ready first into master, then merging master into the branch that remains open.  

Is this ready for review?

brianamarie commented:
> @brianamarie the first step getting stuck is not unique to this course. @hectorsector and @JasonEtco have explained that it has something to with the draft not syncing as we expect. It is happening in other courses as well.

Gotcha. Should I wait to try and go through the course on this branch until this is resolved?

mattdavis0351 commented:
@brianamarie it is resolved, feel free to go through it.  @hectorsector yes, ready for review

brianamarie commented on responses/05_create-metadata.md:
```diff
@@ -15,25 +13,23 @@ We will start with using the parameters that are **required** and later implemen
+   </details>
 
 5. Save the `action.yml` file
 6. commit the changes:
```
```suggestion
6. Commit the changes:
```

brianamarie commented on responses/05_create-metadata.md:
```diff
@@ -15,25 +13,23 @@ We will start with using the parameters that are **required** and later implemen
-   `git commit -m 'update action.yml'`
+   `git add action.yml`
+   `git commit -m 'create action.yml'`
 7. push them to the `hello-world` branch:
```
```suggestion
7. Push them to the `hello-world` branch:
```

brianamarie commented on responses/06_main-js.md:
```diff
@@ -1,35 +1,32 @@
-4. commit the changes:
-   `git add .`
+1. Save the `main.js` file
+1. commit the changes:
```
```suggestion
1. Commit the changes:
```

brianamarie commented on responses/06_main-js.md:
```diff
@@ -1,35 +1,32 @@
+   `git add main.js`
    `git commit -m 'create main.js'`
-5. push them to the `hello-world` branch:
+1. push them to the `hello-world` branch:
```
```suggestion
1. Push them to the `hello-world` branch:
```

brianamarie commented on responses/10_create-metadata.md:
```diff
@@ -32,7 +32,7 @@ We will not use the `joke-output` in in this portion of the course. There will b
 
 6. Save the `action.yml` file
 7. commit the changes:
```
```suggestion
7. Commit the changes:
```

brianamarie commented on responses/10_create-metadata.md:
```diff
@@ -32,7 +32,7 @@ We will not use the `joke-output` in in this portion of the course. There will b
-   `git add .`
+   `git add action.yml`
    `git commit -m 'add metadata for the joke action'`
 8. push them to the `action-two` branch:
```
```suggestion
8. Push them to the `action-two` branch:
```

brianamarie commented on responses/14_create-metadata.md:
```diff
@@ -35,11 +35,11 @@ We will use the `joke-output`, as well as an issue title, in in this portion of
-7. Lastly, define the `run` parameter to use `"node12"` to execute the `"main.js"`
+7. Lastly, define the `runs:` parameter to use `"node12"` and set the `main:` parameter to `"main.js"`
 8. Save the `action.yml` file
 9. commit the changes:
```
```suggestion
9. Commit the changes:
```

brianamarie commented on responses/14_create-metadata.md:
```diff
@@ -35,11 +35,11 @@ We will use the `joke-output`, as well as an issue title, in in this portion of
-   `git commit -m 'update action.yml'`
+   `git add action.yml`
+   `git commit -m 'create action.yml'`
 10. push the changes to the `action-three` branch:
```
```suggestion
10. Push the changes to the `action-three` branch:
```

brianamarie commented on responses/dev-env.md:
```diff
@@ -38,7 +38,7 @@ Now that you have all the necessary tools installed locally, follow these steps
    `git add .`
-   `git commit -m 'initial hello-world'`
+   `git commit -m 'add project dependencies'`
 10. Push you changes to your repository:
```
```suggestion
10. Push your changes to your repository:
```

brianamarie commented on responses/09_new-action.md:
```diff
@@ -7,6 +7,8 @@ Before we continue we are going to need to do a few things. First and foremost o
 1. [Edit]({{workflowFile}}) your workflow file by commenting out every single line.
```
```suggestion
1. [Edit]({{workflowFile}}) your workflow file by commenting out every single line. _(To comment in `.yml`, add a `#` symbol to the beginning of every line)_.
```

brianamarie commented on responses/10_action-two.md:
```diff
@@ -1,33 +1,34 @@
-1. Open the **Terminal** (Mac and Linux) or **Command Prompt** (Windows) on your local machine and navigate to your course repo directory.
-2. Checkout the `master` branch
+1. Open the **Terminal** (Mac and Linux) or **Command Prompt** (Windows) on your local machine
+1. navigate to the `.github/` directory.
```
```suggestion
1. Navigate to the `.github/` directory.
```

brianamarie commented on responses/10_action-two.md:
```diff
@@ -1,33 +1,34 @@
+1. Install the **request**, **request-promise** and **@actions/core** dependencies using `npm`:
    `npm install --save request request-promise @actions/core`
-9. Commit those newly added files,we will remove the need to upload **node_modules** in a later step:
+1. Commit those newly added files,we will remove the need to upload **node_modules** in a later step:
```
```suggestion
1. Commit those newly added files. We will remove the need to upload **node_modules** in a later step:
```

brianamarie committed:
capitalize Try at start of sentence

I tried to do a suggested edit, but it wouldn't let me, so I just went ahead and did this.

brianamarie reviewed (commented):
This is SO AWESOME @mattdavis0351!!!! 

I put in a few suggested changes for little capitalization type things. After those are addressed, I say 🚢!

brianamarie commented:
Ah! Actually, something did go slightly wrong at the very last second. It looks like the last respond has an issue with identifying where it's supposed to be written. You can see the debug issue that was opened in my repository here: https://github.com/brianamarie/writing-javascript-actions/issues/11#issuecomment-578786633

mattdavis0351 committed:
Update responses/05_create-metadata.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 committed:
Update responses/05_create-metadata.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 committed:
Update responses/06_main-js.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 committed:
Update responses/06_main-js.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 committed:
Update responses/14_create-metadata.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 committed:
Update responses/10_create-metadata.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 committed:
Update responses/dev-env.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 committed:
Update responses/10_action-two.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 committed:
Update responses/10_action-two.md

Co-Authored-By: Briana Swift <brianamarie@github.com>

mattdavis0351 committed:
Update responses/09_new-action.md

Co-Authored-By: Briana Swift <brianamarie@github.com>
